src/components/CollectionShare.tsx

```diff
@@ -293,6 +475,146 @@ function CollectionShare(props: Props) {
                 {publicShareUrl ? (
                     <div style={{ width: '100%', wordBreak: 'break-all' }}>
                         <CodeBlock key={publicShareUrl} code={publicShareUrl} />
+
+                        <div
+                            style={{
+                                height: '1px',
+                                marginTop: '10px',
+                                marginBottom: '18px',
+                                background: '#444',
+                                width: '100%',
+                            }}
+                        />
+                        <h6 style={{ marginTop: '2px' }}>
+                            {constants.MANAGE_LINK}
+                        </h6>
+                        <FlexWrapper
+                            style={{ paddingTop: '5px', color: '#fff' }}>
+                            <div style={{ marginRight: '20px' }}>
+                                {constants.FILE_DOWNLOAD}{' '}
+                            </div>
+                            <Form.Switch
+                                style={{ marginLeft: '20px' }}
+                                checked={
+                                    publicShareProp?.enableDownload ?? false
+                                }
+                                id="public-sharing-file-download-toggler"
+                                className="custom-switch-md"
+                                onChange={handleFileDownloadSetting}
+                            />
+                        </FlexWrapper>
+                        <FlexWrapper
+                            style={{ paddingTop: '15px', color: '#fff' }}>
+                            <div style={{ marginRight: '10px' }}>
+                                {' '}
+                                {constants.LINK_DEVICE_LIMIT} :{' '}
+                            </div>
+                            <div style={{ minWidth: '80px' }}>
+                                <Select
+                                    options={downloadOptions}
```

Comment:
Yep. I went ahead with the list here to be consistent with mobile. On mobile, initially, we were using a text box but Vishnu changed it to a selectable list (assuming because we have a small limit: 50)